components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js

```diff
@@ -0,0 +1,524 @@
+              dataRow.remove();
+              var element = deleteTrigger.up('tr').select('.variant-row-count.variant')[0];
+              // Get inner text in cross-browser manner
+              var innerText = element.textContent || element.innerText;
+              var geneCount = innerText.substring(0, innerText.indexOf('.'));
+              var i = 1;
+              $$('.variant-hide-heading-' + geneIndex + ' .variant-row-count').each(function(item) {
+                item.childNodes[1].textContent = geneCount + '.' + (i++);
+              });
+              //-update variant table header count
+              var newHeader = $$('.variant.variant-title.gene-'+geneIndex)[0].firstChild.textContent.replace(/(\().+?(\))/g, "$1"+(--i)+"$2");
+              $$('.variant.variant-title.gene-'+geneIndex)[0].firstChild.textContent = newHeader;
+            }
+          }
+        },
+
+        onComplete : function() {
+          deleteTrigger.disabled = false;
+        }
+
+      },
+      {
+         confirmationText : "$!services.localization.render('phenotips.tableMacros.rowDeleteConfirmation')"
+      });
+    },
+
+    ajaxAddVariantData : function (event) {
+      event.stop();
+
+      var addTrigger = event.element();
+      if (addTrigger.disabled)
+         return;
+      var classname = '';
+      try {
+        var classname = /classname=([^&]+)/.exec(addTrigger.href)[1];
+      } catch (err) {
+         new XWiki.widgets.Notification("$services.localization.render('phenotips.tableMacros.typeNotFound')", 'error');
+         return;
+      }
+
+      var className = addTrigger.up('td.variant').className;
+      var geneIndex = className.substring(className.lastIndexOf('-')+1);
+      var geneSymbol = $$('.gene.col-label.gene-' + geneIndex)[0].innerHTML;
+
+      //lock the genesymbol input field
+      $$('[name="'+this.geneClassName+'_' + geneIndex + '_gene"]')[0].toggleClassName('v-collapsed', true);
+      $$('.gene.col-label.gene-' + geneIndex + '.gene-input-label')[0].toggleClassName('v-collapsed', false);
+
+      var variantFooter = addTrigger.up('tr.variant-footer');
+      if (!variantFooter) {
+        new XWiki.widgets.Notification("$services.localization.render('phenotips.tableMacros.listNotFound')" + ' ' + classname, 'error');
+      }
+
+      var varCount = 1;
+      var geneCount = parseInt($$('td.gene.gene-'+geneIndex)[0].up('tr').down('td').innerHTML, 10);
+      var varIndex = 0;
+      $$('.variant-moreinfo-editbutton-row').each( function(item) {
+        var indx = parseInt(item.className.substring(item.className.lastIndexOf('-')+1), 10);
+        varIndex = Math.max(indx, varIndex);
+      });
+      varIndex++;
+
+
+      var data = {GENE_SYMBOL_PLACEHOLDER: geneSymbol, VARIANT_COUNT_PLACEHOLDER: varCount, GENE_INDEX_PLACEHOLDER: geneIndex, VARIANT_INDEX_PLACEHOLDER: varIndex}
+
+      //If we insert first row of variants - create headers
+      if (variantFooter.previous().className == '') {
```

Comment:
you should use `===` when comparing strings, since `''` can be interpreted as `false` with `==`.